charts/llm-d-modelservice/templates/decode-deployment.yaml

```diff
@@ -23,8 +23,14 @@ spec:
         {{- toYaml .Values.decode.podAnnotations | nindent 8 }}
       {{- end }}
     spec:
+      {{- if or (hasKey .Values.decode "initContainers") (not (hasKey .Values "routing")) (not (hasKey .Values.routing.proxy "enabled")) (ne .Values.routing.proxy.enabled false) }}
```

Comment:
A bit confused what's going on here. Is this valid?

Checks whether _any_ of the following are true:
1. Checks if .Values.decode contains the key "initContainers".
2. Checks if .Values does not contain the key "routing".
3. Checks if .Values.routing.proxy does not contain the key "enabled".
4. Checks if .Values.routing.proxy.enabled is not equal to false (== .Values.routing.proxy.enabled=true?)

Not really sure why 2 and 3 are needed? Also the double negative for 4 is confusing.

Reply:
For some reason I thought I needed to check that these keys exist. However, this does not seem to be the case. I have removed 2 and 3.  Agree on 4. But it does need to be explicitly set to false but not when true. So safer to test false.